test_conformance/images/clCopyImage/test_copy_generic.cpp

```diff
@@ -158,23 +173,40 @@ cl_mem create_image( cl_context context, cl_command_queue queue, BufferOwningPtr
         }
     }
 
-    img = clCreateImage(context, mem_flags, imageInfo->format, &imageDesc, host_ptr, error);
+    if (imageInfo->type != CL_MEM_OBJECT_IMAGE1D_BUFFER)
+    {
+        img = clCreateImage(context, mem_flags, imageInfo->format, &imageDesc,
+                            host_ptr, error);
+    }
+    else
+    {
+        img = clCreateImage(context, mem_flags, imageInfo->format, &imageDesc,
+                            nullptr, error);
+    }
 
     if (gEnablePitch)
     {
+        pitch_buffer_data *data =
+            static_cast<pitch_buffer_data *>(malloc(sizeof(pitch_buffer_data)));
+        data->buf = host_ptr;
+        data->is_aligned = (CL_VERSION_MAJOR(version) != 1)
+            && (imageInfo->type == CL_MEM_OBJECT_IMAGE1D_BUFFER);
```

Comment:
Consider setting up `is_aligned` closer to the call to `align_malloc`.  This might simplify the code and make it easier to handle potential additional calls to `align_malloc` in the future.

Reply:
I added `is_host_pointer_aligned` variable to track which allocation function is used.